src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -63,6 +63,17 @@ protected PlatformVersionInfo GetAvailableVersionsFromStorage(string platformNam
                 ? SdkStorageConstants.LegacySdkVersionMetadataName
                 : SdkStorageConstants.SdkVersionMetadataName;
 
+            // if <NextMarker> element's value is not empty, we iterate through every page by appending marker value to the url
+            // and consolidate blobs from all the pages.
+            do
+            {
+                url = string.Format(SdkStorageConstants.ContainerMetadataUrlFormat, sdkStorageBaseUrl, platformName, marker);
```

Comment:
Nit: should we consider putting this logic in an helper, so that we don't repeat it?